### PR TITLE
Extend the Parquet writer's dictionary encoding benchmark.

### PR DESCRIPTION
**Pull request by mhaseeb123**
## Description
This PR extends the data cardinality and run length range for the existing parquet writer's encoding benchmark.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/io/parquet/parquet_writer.cpp [hunk 1]
```diff
@@ -202,8 +202,8 @@ NVBENCH_BENCH_TYPES(BM_parq_write_encode, NVBENCH_TYPE_AXES(d_type_list))
   .set_name("parquet_write_encode")
   .set_type_axes_names({"data_type"})
   .set_min_samples(4)
-  .add_int64_axis("cardinality", {0, 1000})
-  .add_int64_axis("run_length", {1, 32});
+  .add_int64_axis("cardinality", {1, 1000, 10'000, 100'000, 1'000'000})
```
Comment (mhaseeb123) [inline cpp/benchmarks/io/parquet/parquet_writer.cpp:205]: Are there any resource/infrastructure constraints that we need to consider here? We can perhaps remove one or two entries from here if needed.

**Comment by vuule** [comment]
what's the reason for this change?

**Comment by mhaseeb123** [comment]
> what's the reason for this change?

First of all welcome back. Greg wanted me to push any updates I did to the benchmark for #16541. Though I think that all my local changes (extended ranges mostly) need not to be pushed upstream if not needed.

File: cpp/benchmarks/io/parquet/parquet_writer.cpp (hunk 1 above)
Comment (vuule) [reply cpp/benchmarks/io/parquet/parquet_writer.cpp:205]: I think we can drop `1`. Given the table size, it's effectively the same as 1000.
Also, should 1M just be 0? Not sure if the goal is to have unique elements (AFAIK row count is typically lower than 1M).

File: cpp/benchmarks/io/parquet/parquet_writer.cpp [hunk 2]
```diff
@@ -202,8 +202,8 @@ NVBENCH_BENCH_TYPES(BM_parq_write_encode, NVBENCH_TYPE_AXES(d_type_list))
   .set_name("parquet_write_encode")
   .set_type_axes_names({"data_type"})
   .set_min_samples(4)
-  .add_int64_axis("cardinality", {0, 1000})
-  .add_int64_axis("run_length", {1, 32});
+  .add_int64_axis("cardinality", {0, 1000, 10'000, 100'000})
+  .add_int64_axis("run_length", {1, 32, 64});
```
Comment (vuule) [inline cpp/benchmarks/io/parquet/parquet_writer.cpp:206]: Sorry, missed this the first time around. To me, 32 is already a very high run length. I think we should instead add 4 or 8. I'm fine with 64 if you do see a significant difference in file size and/or performance compared to 32.

Comment (mhaseeb123) [reply cpp/benchmarks/io/parquet/parquet_writer.cpp:206]: No significant differences with 64 for the benchmarks we ran either.

Comment (vuule) [reply cpp/benchmarks/io/parquet/parquet_writer.cpp:206]: I vote for 4 or 8, then

Comment (mhaseeb123) [reply cpp/benchmarks/io/parquet/parquet_writer.cpp]: All done!

**Review by vuule — APPROVED** [review]

**Review by karthikeyann — APPROVED** [review]
Would be nice to know how much time this increases in benchmark runs.
If it is not available now, follow up with Randy on benchmark runs.

**Comment by mhaseeb123** [comment]
> Would be nice to know how much time this increases in benchmark runs. If it is not available now, follow up with Randy on benchmark runs.

Results in #16541 for which we are extending this.

**Pull request merged**